lib/cashbox-batches.js

```diff
@@ -18,6 +19,32 @@ function createCashboxBatch (deviceId, cashboxCount) {
   })
 }
 
+function updateMachineWithBatch (machineContext, oldCashboxCount) {
+  if (!_.has(['deviceId', 'cashbox', 'cassettes'], machineContext) && !_.inRange(constants.CASH_OUT_MINIMUM_AMOUNT_OF_CASSETTES, constants.CASH_OUT_MAXIMUM_AMOUNT_OF_CASSETTES + 1, _.size(machineContext.cassettes)))
+    throw new Error('Insufficient info to create a new cashbox batch')
+  if (_.isEqual(0, oldCashboxCount)) throw new Error('Cashbox is empty. Cashbox batch could not be created.')
+
+  return db.tx(t => {
+    const deviceId = machineContext.deviceId
+    const batchId = uuid.v4()
+    const q1 = t.none(`INSERT INTO cashbox_batches (id, device_id, created, operation_type) VALUES ($1, $2, now(), 'cash-in-empty')`, [batchId, deviceId])
+    const q2 = t.none(`UPDATE bills SET cashbox_batch_id=$1 FROM cash_in_txs
+      WHERE bills.cash_in_txs_id = cash_in_txs.id AND
+      cash_in_txs.device_id = $2 AND 
+      bills.cashbox_batch_id IS NULL`, [batchId, deviceId])
+    const q3 = t.none(`UPDATE devices SET cashbox=$1, cassette1=$2, cassette2=$3, cassette3=$4, cassette4=$5 WHERE device_id=$6`, [
+      machineContext.cashbox,
+      machineContext.cassettes[0],
+      machineContext.cassettes[1],
+      machineContext.cassettes[2],
+      machineContext.cassettes[3],
```

Comment:
Do we have 0 as a default value for each cassette? For devices that only have 2 for instance